### PR TITLE
[RLlib] Examples folder restructuring (models) part 1

### PR DESCRIPTION
**Pull request by sven1977**
Follow up PR for #8250 . 

This PR cleans up the rllib/examples folder by copying(!) (moving will be done in follow up PR to keep this PR simpler) all example ModelV2s into rllibexamples/models, so they can be used by other scripts and tests as well. Also, PyTorch ModelV2-pendants have been created and placed (together with their tf-counterparts) into the new example/models folder.

This is in preparation of reaching 100% tf-torch parity in this folder.

In a third follow-up PR, we will make sure, all examples are tested in both frameworks as well as learn the given task (this is often currently not checked).

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: rllib/agents/dqn/dqn_torch_model.py [hunk 1]
```diff
@@ -87,9 +87,9 @@ def __init__(
             advantage_module.add_module("A", nn.Linear(ins, action_space.n))
             value_module.add_module("V", nn.Linear(ins, 1))
         # Non-dueling:
-        # Q-value layer (use Advantage module's outputs as Q-values).
+        # Q-value layer (use main module's outputs as Q-values).
```
Comment (sven1977) [inline rllib/agents/dqn/dqn_torch_model.py:90]: This was actually a torch bug in DQN, where the output was once again pushed through yet another Linear. We do not do this in tf and should not do this in torch either, of course.

File: rllib/examples/env/rock_paper_scissors.py [hunk 2]
```diff
@@ -23,6 +23,9 @@ def __init__(self, config):
         self.last_move = None
         self.num_moves = 0
 
+        # For test-case inspections (compare both players' scores).
```
Comment (sven1977) [inline rllib/examples/env/rock_paper_scissors.py:26]: This allows us to inspect for actual learning progress of one of the players (winning over the other). Rewards are always 0.0 (zero-sum game).

File: rllib/examples/models/autoregressive_action_dist.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
+from ray.rllib.models.tf.tf_action_dist import Categorical, ActionDistribution
```
Comment (sven1977) [inline rllib/examples/models/autoregressive_action_dist.py:1]: All example Models were copied (will be moved in follow-up PR) into the examples/models/ dir and a PyTorch version to each one was added, such that all example scripts will be supporting a `--torch` flag to run the exact same script using PyTorch.

File: rllib/models/catalog.py [hunk 4]
```diff
@@ -281,14 +281,16 @@ def get_model_v2(obs_space,
             model_cls = _global_registry.get(RLLIB_MODEL,
                                              model_config["custom_model"])
             if issubclass(model_cls, ModelV2):
+
```
Comment (sven1977) [inline rllib/models/catalog.py:284]: Bug fix: Wrapping was not done properly for Torch Models.

File: rllib/utils/test_utils.py [hunk 5]
```diff
@@ -225,3 +225,21 @@ def check(x, y, decimals=5, atol=None, rtol=None, false=False):
                 if false is True:
                     assert False, \
                         "ERROR: x ({}) is the same as y ({})!".format(x, y)
+
+
+def check_learning_achieved(tune_results, min_reward):
```
Comment (sven1977) [inline rllib/utils/test_utils.py:230]: A call to this will be added to each example script and executed if an `--as-test` flag. It'll then check, whether some learning goal (reward) was reached in the example script. We currently do NOT test our example scripts for learning success.

**Review by ericl — APPROVED** [review]

File: rllib/policy/torch_policy.py [hunk 6]
```diff
@@ -232,7 +232,7 @@ def learn_on_batch(self, postprocessed_batch):
         loss_out = force_list(
             self._loss(self, self.model, self.dist_class, train_batch))
         assert len(loss_out) == len(self._optimizers)
-        assert not any(np.isnan(l.detach().numpy()) for l in loss_out)
+        # assert not any(np.isnan(l.detach().numpy()) for l in loss_out)
```
Comment (ericl) [inline rllib/policy/torch_policy.py:235]: Remove instead of commenting out?

Comment (sven1977) [reply rllib/policy/torch_policy.py:235]: +1
I would have liked to keep it in, but it breaks some dummy-agent test cases.
